tests/services/order_validation_service/test_case_sample_rules.py

```diff
@@ -48,3 +50,55 @@ def test_validate_tube_container_name_unique(valid_order: OrderWithCases):
     # THEN the error should concern the non-unique tube container name
     assert isinstance(errors[0], ContainerNameRepeatedError)
     assert errors[0].sample_index == 0 and errors[0].case_index == 0
+
+
+def test_invalid_required_volume(valid_order: OrderWithCases):
+
+    # GIVEN an orders with two sample with invalid required volume, and different container
+    valid_order.cases[0].samples[0].container = ContainerEnum.tube
+    valid_order.cases[0].samples[0].volume = None
+
+    valid_order.cases[0].samples[1].container = ContainerEnum.plate
+    valid_order.cases[0].samples[1].volume = None
+
+    # WHEN validating the two orders
+    errors = validate_case_required_volume(order=valid_order)
+
+    # THEN an error should be returned
+    assert errors
+
+    # THEN the error should concern the invalid sample index
+    assert isinstance(errors[0], VolumeRequiredCaseError)
+    assert errors[0].sample_index == 0 and errors[0].case_index == 0
+
+    assert isinstance(errors[1], VolumeRequiredCaseError)
+    assert errors[1].sample_index == 1 and errors[1].case_index == 0
+
+
+def test_valid_required_volume(valid_order: OrderWithCases):
```

Comment:
This test is likely not needed since there is an integration test for a valid TomteOrder which is an OrderWithCases.

Reply:
Hmm, so I wouldn’t need to test it at all, or should I use a different test set?

Reply:
Hopefully I’m not spamming you with too many messages, but could you @islean help me understand this?

Reply:
Exactly - I think we should have one test which performs all validations for whole valid order. Then we do not need tests which make sure that each individual validation passes for a valid order.

Might need to set up the order a bit intricately to test all the caveats but in principle I think it is a better approach. This test won't cause any issues but adding a "valid order" test for each rule might make the test files a bit rough to read.

Reply:
Yes, that makes so much sense—I hadn’t thought of it that way.

Reply:
I created a new test that validates all the current rules against a case with valid values. Please let me know what you think.

Reply:
Sorry, I was a bit unclear in what I meant. There is already such a test for Tomte, check out `test_valid_order(valid_order: TomteOrder, tomte_validation_service: TomteValidationService)` in `test_validation_integration.py` file. We have a method in all `OrderValidationServices` called `_get_errors` which runs all the validation rules for the specified workflow and returns any errors. We should probably have one per workflow but just by setting your rules to be included in the Tomte validation rules, this is already taken care of.